files/usr/share/cinnamon/applets/network@cinnamon.org/applet.js

```diff
@@ -1692,6 +1700,10 @@ CinnamonNetworkApplet.prototype = {
 
             this.settings = new Settings.AppletSettings(this, metadata.uuid, this.instance_id);
             this.settings.bind("keyOpen", "keyOpen", this._setKeybinding);
+            this.settings.bind("num-visible-networks", "num_visible_networks");
+            this.settings.bind("hide-other-networks-when-connected", "hide_other_networks_when_connected");
+            APPLET_SETTINGS.num_visible_networks = this.num_visible_networks;
+            APPLET_SETTINGS.hide_other_networks_when_connected = this.hide_other_networks_when_connected;
```

Comment:
These settings won't ever get updated - you'd need a cinnamon restart to make any changes apply.  I believe you can provide a callback to the .bind calls, then update APPLET_SETTING's values then.